internal/init/init.go

```diff
@@ -91,34 +92,49 @@ func (c *Client) NewCommand() *cobra.Command {
 			c.UserService = *metalGoClient.UsersApi
 			c.ProjectService = *metalGoClient.ProjectsApi
 
-			user, _, err := c.UserService.FindCurrentUser(context.Background()).Execute()
+			var defaultOrgId, defaultProjectId string
+			project, isProjectToken, err := getDefaultProjectOK(c.ProjectService)
```

Comment:
If the token is a user token, this has the potential to be extremely slow, because we're going to load every single project the user has access to before we look at the user's profile (which might already have a default project ID).  To reduce the impact on runtime for existing use cases, we could instead (pseudocode):

```
user, err := FindCurrentUser()

if err != nil {
  // I don't know what this error looks like; we might be able to make smart decisions based on error message
  projects, err := GetAllProjects()
  if err != nil {
    return err
  }
  if length(projects) > 1 {
    return errors.New("Failed to load user but got more than one project, I don't know what kind of token this is")
  } else {
    // Do project token setup
  }
}

// Successfully loaded current user, do user token setup
}
```

Reply:
This PR needs to be rebased, as the codebase and its dependencies have changed quite a bit in the meantime (for example, the SDK now provides pagination helpers, so we don't need to (and shouldn't) use the `GetAllProjects` function), but in general the pseudocode above should allow us to add support for project tokens without drastically increasing the startup time when someone is using a user token.